### PR TITLE
feat: filter barracuda

### PR DESCRIPTION
**Pull request by nandinivij**
There is no host here. Checked avi_vantage_no_host and worked. Please review

### CODE REVIEW

File: tests/test_barracuda.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    host = "{}-{}".format(random.choice(setup_wordlist), random.choice(setup_wordlist))
+
+    dt = datetime.datetime.now()
+    iso, bsd, time, date, tzoffset, tzname, epoch = time_operations(dt)
+
+    # Tune time functions
+    epoch = epoch[:-3]
+
+    mt = env.from_string(
+        '{{ mark }} {{ iso }} {{ host }} Barracuda - - - - [meta sequenceId="2376780"]src=10.1.1.1 spt=33217 dst=10.1.1.1 dpt=39971 actionTaken=DENY attackDescription=GEO_IP_BLOCK attackDetails=GeoIP Policy Match attackGroup=Forceful Browsing attackId=1111 logType=WF app=TLSv1.2 request=/apps/ requestMethod=GET rt=1649197620642 userAgent=Mozilla/5.0 [en] (X11, U; OpenVAS-VT 9.0.3) referer='
```
Comment (rfaircloth-splunk) [inline tests/test_barracuda.py:26]: there are one to many "- "
remove [meta sequenceId="2376780"] this is an artifact from a syslog-ng bug this isn't in the source

File: package/etc/conf.d/conflib/syslog/app-syslog-barracuda_syslog.conf [hunk 2]
```diff
@@ -0,0 +1,36 @@
+block parser app-syslog-barracuda_syslog() {
+ channel {
+        rewrite {
+            r_set_splunk_dest_default(
+                index("netfw")
+                sourcetype('barracuda')
+                vendor("barracuda")
+                product("waf")
+                template("t_hdr_msg")
```
Comment (rfaircloth-splunk) [inline package/etc/conf.d/conflib/syslog/app-syslog-barracuda_syslog.conf:9]: for this one we can use t_msg_only as template there is no data in the header needed

File: docs/sources/vendor/Barracuda/barracuda.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Barracuda
```
Comment (rfaircloth-splunk) [inline docs/sources/vendor/Barracuda/barracuda.md:1]: rename this file "waf.md"

File: docs/sources/vendor/Barracuda/barracuda.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+| sourcetype      | notes                                                                                                   |
+|-----------------|---------------------------------------------------------------------------------------------------------|
+|barracuda:web:firewall   |  none  |
+
+## Sourcetype and Index Configuration
+
+| key             | sourcetype     | index  | notes          |
+|-----------------|----------------|--------|----------------|
+| Barracuda       |  barracuda:web:firewall  | netfw  | None     |
```
Comment (rfaircloth-splunk) [inline docs/sources/vendor/Barracuda/barracuda.md:26]: the key is "vendor_product_class" where _class is optional so the key here should be barracuda_waf

**Review by rjha-splunk — COMMENTED** [review]
made the changes

**Pull request merged**